### ISSUE TITLE
Restructuring of the store

### ISSUE THREAD

**Issue by DSWallach** [jspsych/jsPsych-Redux-GUI#12]
-------- DONE SO FAR --------
Reducers are now split up into multiple files, all in the src/reducers/ directory. Each property of the state of the store (e.g. trialTable, trialOrder, openTrial, etc...) has it's own reducers file name (propertyname)Reducers.jsx. These files are indexed in webpack.config.js, and are all imported into src/reducers/index.jsx.
```
// Combine all the different reducers into the store
const rootReducer = combineReducers({
    trialTable,
    trialOrder,
    openTrial,
    timelineOpen,
    pastStates,
    futureStates,
    dragged,
    over
});
export default rootReducer;
```

This file contains the combined reducer for the store. When an action dispatches a reducer to the store, combineReducers handles passing that reducer along to each property that has a matching reducer. For example when 'ADD_TRIAL' is dispatched, the reducer matching that string is called on both the trialTable prop
```
case 'ADD_TRIAL':
        // New trial's name 
        var newName = 'Trial_' + Object.keys(state).length;
        // Make the new trial from the default template.
        var newTrial = Object.assign({}, Trial);
        // Delete is okay as these shallow copies are not yet part
        // of the state. 
        delete newTrial['name'];
        delete newTrial['id'];
        // Add the new properties
        newTrial['id'] = action.id;
        newTrial['name'] = newName;

        // Create the new trial table
        var newState = Object.assign({}, state);
        newState[index] = newTrial;

        return newState;
```
and the trialOrder prop
```    
case 'ADD_TRIAL':
        var newState = [
            ...state,
            action.id
        ];
        return newState;
```
The reducer for each prop, modifies only that prop, and returns it as its new state. combineReducers then grabs the updated props, and returns the new state with the modified properties.

Since each reducer is responsible for changing only it's property it is import that any logic that is relevant to mutliple reducers (if else statements, determining what trials to remove, etc...) happens in the action that calls the reducer and not in the reducer itself. This will avoid doing the same work multiple times when multiple properties of the store have matching cases for a particular reducer. REMOVE_TRIAL is a good example of this.
-- actionRemoveTrial --
```
// Action calling for a trial to be removed from trialList
// Affects: trialTable, trialOrder
export const actionRemoveTrial = (store) => {
    var state = store.getState();

    // Call back predicate for use by filter
    const isSelected = (trialID) => {
        state.trialTable[trialID].selected;
    };
    // List of trials to be removed
    var removeList = Object.keys(state.trialTable);
    removeList = removeList.filter(isSelected);

    actionArchiveState(store);
    store.dispatch({
        type: 'REMOVE_TRIAL',
        index: state.selected,
        state: state,
        toRemove: removeList
    });
};
``` 
Note how the comments for the action describes which properties of the store are
affected by the reducer this action calls. There may be a better way to indicate this but
I think it's important to indicate it for every action defined so that a programmer looking
at an action will know immediately what properties are affected inside of having to search 
through every reducer's file.

-------- STILL TO DO --------
Identify and fix the errors currently appearing by testing for each property.

**Comment by DSWallach** [comment]
UPDATE: Added unit testing for trialOrder.
Use 'npm test' (defined in 'root/package.json') to run all the tests.
It provides one test for each reducer that affects trialOrder, although more, situational tests will be added.

The setup for combineReducers is fully implemented and GUI renders and the initial state is created correctly. However many of the reducers do not function properly currently. Some of this is due to the change in setup of the state, but as we discovered Thursday some of these issue have been lurking for longer than that.  Since there is now a template for testing the reducers for each property I thought it might be a good time for me to merge this branch back into the master, depending on what you think @jodeleeuw. 

That way @kristiyip can add testing for the reducers you've written while I add the tests for those I've written. Thanks to the new setup these tests can operate completely independently. Each store property should have it's own testing file, based on the template provided by trialOrderTests.jsx. Once all the reducers are tested, we can added similar tests for the actions, as many of them contain internal logic that should also be testing. I expect that once every action/reducer is tested and passes, every in the gui will work properly. 

Let me know if either of you have any questions or if a can proceed with the merge at your earliest convenience.
